Mention LHv2 doesn't support trim with AIO bdev driver (#676)
harvester/docs: docs/advanced/longhorn-v2.md

  :::note

>   Harvester sets the [Longhorn disk driver](https://longhorn.io/docs/1.7.2/v2-data-engine/features/node-disk-support/) to `auto` so that NVMe disks use the SPDK NVMe bdev driver, which provides the best performance and also supports advanced operations like trimming.

Reviewer: ```suggestion
  Harvester sets the [Longhorn disk driver](https://longhorn.io/docs/1.7.2/v2-data-engine/features/node-disk-support/) to `auto` so that NVMe disks use the SPDK NVMe bdev driver, which provides the best performance and also supports advanced operations such as trim (also known as discard).
```

Reviewer: You can change :::note to :::info important (since the text block is no longer purely informational).

>   SSDs and other non-NVMe disks are managed using the SPDK AIO bdev driver, which requires a disk size that is an *even multiple of 4096 bytes*. Non-NVMe disks that do not meet this size requirement cannot be added.

Reviewer: ```suggestion
  SSDs and other non-NVMe disks are managed using the SPDK AIO bdev driver, which requires a disk size that is an *even multiple of 4096 bytes*. Non-NVMe disks that do not meet this size requirement cannot be added.  Additionally, the SPDK AIO bdev driver does not support the unmap operation. If you are using non-NVMe disks, avoid trimming the filesystem because this results in I/O errors and paused virtual machines.
```

>   Additionally, the SPDK AIO bdev driver does not support the unmap operation. This means attempting to use trimming (also known as discard) with these devices will result in I/O errors and will pause your virtual machines. Accordingly, if you are using non-NVMe disks, you should avoid trim/discard.  For example, when creating an ext4 filesystem in a Linux VM, use `mkfs.ext4 -E nodiscard /dev/vdb` (assuming `/dev/vdb` is your device path).  On Windows VMs, trim support for NTFS can be disabled by running `fsutil behavior set disabledeletenotify NTFS 1` from the command prompt.

Reviewer 2: A note seems too gentle, we should raise the level to caution or warning. Or we can integrate this into the "Limitations" section?

Author: Yeah I did wonder if this wanted to be moved to another section. Happily it's not a problem for NVMe disks (i.e. anything that shows up on the host as `/dev/nvmeXXXX`), but folks trying non-NVMe SSDs will likely hit this.

Reviewer: As Kiefer suggested, please move this to the Limitations section and use an unordered list.

- Within the Harvester context, volumes backed by the Longhorn V2 Data Engine must be added to virtual machines as extra disks. The boot disk of each virtual machine must still be added from an image that is backed by the Longhorn V1 Data Engine.

- Harvester is unable to live-migrate virtual machines with V2 volumes attached, which means those virtual machines will be shut down during Harvester cluster upgrades. Moreover, snapshots of V2 volumes cannot be created because snapshot and restoration functionality in Harvester relies on volume cloning.

- SSDs and other non-NVMe disks are managed using the SPDK AIO bdev driver, which does not support the unmap operation. If you are using non-NVMe disks, avoid trimming the filesystem because this results in I/O errors and paused virtual machines. For example, when creating an ext4 filesystem on a Linux virtual machine, use `mkfs.ext4 -E nodiscard /dev/vdb` (assuming `/dev/vdb` is your device path). On Windows virtual machines, you can disable trimming for NTFS by running the command `fsutil behavior set disabledeletenotify NTFS 1`.
